File: crawl/flask/app/model/pwa.py
from .base import db, BaseModel
from datetime import datetime
import enum

class AcceptanceStatusEnum(enum.Enum):
    NONE = "NONE"
    PENDING = "PENDING"
    ACCEPTED = "ACCEPTED"

class PWA(BaseModel):
    __tablename__ = "pwa"

    app_id = db.Column(db.String(255), nullable=False, unique=True)
    icon_image = db.Column(db.String(255))
    website_url = db.Column(db.String(255), nullable=False)  # 메인 페이지
    download_url = db.Column(db.String(255), nullable=False)  # 패키지 다운로드 링크
    avg_score = db.Column(db.Float)
    download_count = db.Column(db.Integer, nullable=False, default=0)
    file_size = db.Column(db.String(31))
    version = db.Column(db.String(31))
    company = db.Column(db.String(255))
    developer_site = db.Column(db.String(255))
    created_at = db.Column(db.DateTime, nullable=False, default=datetime.now)
    updated_at = db.Column(db.DateTime, onupdate=datetime.now)
    blocked_at = db.Column(db.DateTime)
    acceptance_status = db.Column(db.Enum(AcceptanceStatusEnum), default=AcceptanceStatusEnum.NONE)
    accepted_at = db.Column(db.DateTime)

File: crawl/pwandora/models/pwa.js
const Sequelize = require('sequelize');
module.exports = function(sequelize, DataTypes) {
  return sequelize.define('pwa', {
    id: {
      autoIncrement: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    acceptance_status: {
      type: DataTypes.ENUM('ACCEPTED','NONE','REJECTED'),
      allowNull: true,
      defaultValue: "NONE"
    },
    accepted_at: {
      type: DataTypes.DATE(6),
      allowNull: true
    },
    app_id: {
      type: DataTypes.STRING(255),
      allowNull: false,
      unique: "UKel84m865xfo2gk038hpqgi144"
    },
    avg_score: {
      type: DataTypes.FLOAT,
      allowNull: true,
      defaultValue: 0
    },
    blocked_at: {
      type: DataTypes.DATE(6),
      allowNull: true
    },
    company: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    developer_site: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    download_count: {
      type: DataTypes.INTEGER,
      allowNull: true,
      defaultValue: 0
    },
    icon_image: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    version: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    website_url: {
      type: DataTypes.STRING(255),
      allowNull: false
    },
    age_limit: {
      type: DataTypes.ENUM('ALL','EIGHTEEN_PLUS','SEVEN_PLUS','SIXTEEN_PLUS','THREE_PLUS','TWELVE_PLUS','UNRATED'),
      allowNull: true,
      defaultValue: "UNRATED"
    }
  }, {
    sequelize,
    tableName: 'pwa',
    timestamps: true,
    indexes: [
      {
        name: "PRIMARY",
        unique: true,
        using: "BTREE",
        fields: [
          { name: "id" },
        ]
      },
      {
        name: "UKel84m865xfo2gk038hpqgi144",
        unique: true,
        using: "BTREE",
        fields: [
          { name: "app_id" },
        ]
      },
      {
        name: "idx_pwa_search",
        using: "BTREE",
        fields: [
          { name: "acceptance_status" },
          { name: "blocked_at" },
          { name: "created_at" },
        ]
      },
      {
        name: "idx_pwa_acceptance_status",
        using: "BTREE",
        fields: [
          { name: "acceptance_status" },
        ]
      },
    ]
  });
};
